src/Gui/BaseApplication.cpp

```diff
@@ -312,9 +313,16 @@ void BaseApplication::initialize( const WindowFactory& factory ) {
     {
         if ( loadFile( parser.value( "camera" ) ) )
         {
+
+            /// \todo fix this, I don't get this code (d.),
+            /// updated using cameraManager.
             auto entity = *( m_engine->getEntityManager()->getEntities().rbegin() );
-            auto camera = static_cast<Engine::Scene::Camera*>( entity->getComponents()[0].get() );
-            m_viewer->setCamera( camera );
+            auto camera =
+                static_cast<Engine::Scene::CameraComponent*>( entity->getComponents()[0].get() );
+            auto cameraManager = static_cast<Ra::Engine::Scene::CameraManager*>(
+                Engine::RadiumEngine::getInstance()->getSystem( "DefaultCameraManager" ) );
+
```

Comment:
This code set the first loaded camera as active. This seems not needed now as CameraManager could do that. right ?

Reply:
right,

Reply:
The question is how do we now that getComponents()[0] is a camera ?
Does it rely on the file given that contains only a camera ?

Reply:
Well, it's more of a hack but it's actually very clear :p.
File associated to the "camera" parameter are expected to contains only cameras.
So the entity access with `rbegin` that is the last entity added to the Engine is the entity containing the loaded cameras. Then, we are sure that the requested component is a camera.
Nevertheless, I think this kind of hypothesis is at the base of several nightmares one can have when trying to understand  (and fix) some bugs that appear due to such undocumented, non robust and risky operations.